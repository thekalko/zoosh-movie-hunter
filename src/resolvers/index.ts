import { IResolvers } from "graphql-tools";

//queries
import userQueries from "./queries/movieQueries";

//mutations
import userMutations from "./mutations/movieMutations";

//subscription
import userSubscription from "./subscriptions/movieSubscriptions";


export const resolverMap: IResolvers = {
  Mutation: {
    ...userMutations
  },
  Query: {
    ...userQueries
  },
  Subscription: {
    ...userSubscription
  }
};