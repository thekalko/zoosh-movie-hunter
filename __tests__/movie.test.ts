import { createTestClient } from 'apollo-server-testing';
import { ApolloServer } from 'apollo-server-express';

import { resolverMap } from '../src/resolvers';
import typeDefs from '../src/schemas';
import { makeExecutableSchema } from 'graphql-tools';

const createTestServer = () => {
  return createTestClient(new ApolloServer({
    schema: makeExecutableSchema({
      typeDefs: typeDefs,
      resolvers: resolverMap,
    }),
    tracing: false,
    cacheControl: false,
    debug: true,
    introspection: true,
    playground: true,
  }));
}

describe('movies test', () => {
  it('fetch movies', async () => {

    const {query} = createTestServer();
  
    const MOVIES_QUERY = `query { 
      movies { 
        title
      }
    }`;
  
    const res = await query({ query: MOVIES_QUERY});
    expect(res).toMatchSnapshot();
  });
})

