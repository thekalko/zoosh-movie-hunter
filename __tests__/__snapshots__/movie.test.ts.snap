// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies test fetch movies 1`] = `
Object {
  "data": Object {
    "movies": Array [
      Object {
        "title": "Gabriel's Inferno Part II",
      },
      Object {
        "title": "Gabriel's Inferno",
      },
      Object {
        "title": "The Shawshank Redemption",
      },
      Object {
        "title": "The Godfather",
      },
      Object {
        "title": "Dilwale Dulhania Le Jayenge",
      },
      Object {
        "title": "Dedicated to my ex",
      },
      Object {
        "title": "Schindler's List",
      },
      Object {
        "title": "Your Name.",
      },
      Object {
        "title": "The Godfather: Part II",
      },
      Object {
        "title": "Spirited Away",
      },
      Object {
        "title": "Parasite",
      },
      Object {
        "title": "The Green Mile",
      },
      Object {
        "title": "My Hero Academia: Heroes Rising",
      },
      Object {
        "title": "Pulp Fiction",
      },
      Object {
        "title": "Hamilton",
      },
      Object {
        "title": "The Lord of the Rings: The Return of the King",
      },
      Object {
        "title": "Clouds",
      },
      Object {
        "title": "Life Is Beautiful",
      },
      Object {
        "title": "The Dark Knight",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
